clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -482,6 +484,14 @@ public long nextHeartbeatMs(final long currentTimeMs) {
             return heartbeatTimer.remainingMs();
         }
 
+        public void onFailedAttempt(final long currentTimeMs) {
+            // Expire timer to allow sending HB after a failure. After a failure, a next HB may be
+            // needed with backoff (ex. errors that lead to retries, like coordinator load error),
+            // or immediately (ex. errors that lead to rejoining, like fencing errors).
+            heartbeatTimer.update(heartbeatTimer.currentTimeMs() + heartbeatTimer.remainingMs());
```

Comment:
What about adding a method to `Timer` that expires the timer without updating it with a time point in the future. Alternatively, I think you could reset the `Timer` to 0 with `heartbeatTimer.reset(0)`.
Do we need a verification in the unit tests for this?

Reply:
Done, good point. I changed it to reset to 0, that shows the intention of not having an interval to wait for, which is what we want on these failure scenarios.

Reply:
Your change to `heartbeatTimer.reset(0)` is not totally equivalent  to `heartbeatTimer.update(heartbeatTimer.currentTimeMs() + heartbeatTimer.remainingMs());` because in the former the `heartbeatTimer` is not expired until `heartbeatTimer.update()` is called, whereas in the latter the heartbeat is expired after the call, but I think in this specific case it does not matter. Is my assumption correct?

Reply:
Sorry, my previous comment is not correct. The `heartbeatTimer` is expired when you call `heartbeatTimer.reset(0)`.